packages/nimble-components/src/combobox/specs/IxD.md

```diff
@@ -0,0 +1,241 @@
+| ----------- | ----------------------------------------------------------------------- |
+| Label       | Text that tells the end-user what to expect in the component            |
+| Field input | Input that displays the current selected or entered option              |
+| Menu button | Click area that opens the menu                                          |
+| Menu        | A list of options to choose from displayed as an _open_ state           |
+| Menu item   | A selection the end-user can make, shown with other options in the menu |
+
+#### Label
+
+Comboboxes should always have a label, unless approved by an interaction designer or user researcher (usually in toolbar use cases). _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Field Input
+
+Comboboxes can be empty by default or have a default selection from the menu. Empty comboboxes should include placeholder text that propts the end-user to enter a custom value or select an option from the menu. Placeholder text should always follow the pattern "Enter or select [thing(s)]", for example "Enter or select address". Ellipses are not needed. Use sentence casing.
+
+![ ](spec-images/combobox-2.png)
+
+#### Menu & Menu Items
+
+A list of options to select, displayed as an open state._See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+### Related Components
+
+To allow end-users to select a menu item but not enter a custom value, use a [**select**](/packages/nimble-components/src/select/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-select.png)
+
+To allow end-users to multi-select menu items, use a [**tag picker**](/packages/nimble-components/src/tag-picker/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-picker.png)
+
+## Design
+
+### Configuration
+
+#### Label
+
+The client-user should be able to specify where they want the label to be positioned. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Icons
+
+The client-user should be able to specify whether they want left icons on menu items or not. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Groups
+
+The client-user should be able to create groups for menu items. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client should be able to specify whether they want the group name to be visible on the component when the menu is closed.
+
+![ ](spec-images/combobox-26.png)
+
+#### Secondary Text
+
+The client-user should be able to specify whether they want secondary text on menu items or not. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Prefix & Suffix
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify a prefix and/or suffix that appears in the field.
```

Comment:
This feels like a pretty loaded feature, that I would suggest we move to the 'Future Considerations' section. There are a few concerns I have around this:
- This ultimately feels like a "universal" feature for various inputs to have like the `TextField` and `NumberTextField`. As a result, if we decide such a feature makes sense (which I think has room for debate) then we would need to be deliberate about its API
- There are some potential localization concerns particularly if we were promoting usages like putting a '$' as a prefix. Some currency types wouldn't put the symbol at the front, but at the end, however using the "suffix API" to do it might be inadequate because of spacing.
- Should we instead just promote a more general slotting API, where users can slot something at either the front and/or the end? While we lose some styling control here, it offers greater flexibility and avoids localization issues

Reply:
I wasn't trying to be prescriptive with the prefix/suffix content, the image is just an example of what a prefix and suffix might look like. I was thinking of this as a general slotting API for the front and the end, and those slots have a specific visual design that distinguishes them from the displayed value. And agreed, there are other input components that would benefit from this feature.

Reply:
I'd like to push this to future work unless we have specific known expected use cases today, i.e. this is used in exisiting designs. If so some examples would help because I also feel like this is a bit of a can of worms.